theano/tests/unittest_tools.py

```diff
@@ -244,9 +244,9 @@ def _compile_and_check(self, inputs, outputs, numeric_inputs, cls,
         # Check that the Op is removed from the compiled function.
         if check_topo:
             topo_shape = shapes_function.maker.fgraph.toposort()
+            topo_out = outputs_function.maker.fgraph.toposort()
             assert not any(isinstance(t.op, cls) for t in topo_shape)
-        topo_out = outputs_function.maker.fgraph.toposort()
-        assert any(isinstance(t.op, cls) for t in topo_out)
```

Comment:
What the docstring meant, even though it is not clear, is that if `check_topo` is True, it will check that the original Op is not (any more) in the graph that computes its shape only.
However, when we want to compare the result of the `infer_shape` optimizations with the original Op, we _need_ the original Op in the graph to make sure we are comparing the right thing.

The way to resolve this here, in my opinion, would be to revert that change, and instead call `self._compile_and_check(..., excluding=inline_ofg_expansion)`, so the original OpFromGraph can be executed.